recog/src/main/java/com/rapid7/recog/parser/RecogParser.java

```diff
@@ -105,7 +105,11 @@ public RecogMatchers parse(File file)
    */
   public RecogMatchers parse(Reader reader, String name)
       throws ParseException {
-    return parse(reader, null, name);
+    RecogMatchers matchers = parse(reader, null, name);
```

Comment:
This should be in the root parse method `public RecogMatchers parse(Reader reader, String path, String name)` otherwise the other constructors could behave differently.